x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx

```diff
@@ -97,11 +56,17 @@ export const Expressions: React.FC<Props> = props => {
   });
   const [timeSize, setTimeSize] = useState<number | undefined>(1);
   const [timeUnit, setTimeUnit] = useState<TimeUnit>('m');
-
   const derivedIndexPattern = useMemo(() => createDerivedIndexPattern('metrics'), [
     createDerivedIndexPattern,
   ]);
 
+  const chartData = useMetricsExplorerChartData(
+    alertParams,
+    alertsContext,
+    derivedIndexPattern,
+    source
+  );
```

Comment:
Any time I edit a condition, all of the charts reload. It can be pretty distracting. Is there a way to not refresh charts that are not related to the current condition? Do you think it'd be better to load each chart separately? It would be more calls to the backend, but I think the UI would feel better when editing.

Reply:
if we are not concerned about the multiple loads then I think that would be fine. I will make the change and see how it feels.